[GR Phase 2] Step 5 Director information screen (without API calls) - implementation

burczu opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

This PR contains implementation of step 5 (Director information screen) - GR phase 2.

Design doc section: https://docs.google.com/document/d/1Y4P7G9h0Easjcea7wyk52gKrNB9o1yTjpmJFYkrHDW4/edit?tab=t.0#bookmark=id.damtu6jbu35v

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/50901
PROPOSAL: n/a


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Log in to the app
2. Open Workspace with Workflows feature enabled
3. Tap on "Connect bank account" inside the "Make or track payments" section
4. Go to the 5th step of the flow
5. Test the flow

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Same as above.

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
6. Upload an image via copy paste
7. Verify a modal appears displaying a preview of that image
--->

Same as above.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/9323a810-0a99-440c-a1d1-ed55625f436a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/3aedd0fa-53a6-4af1-92e3-ce7ecd964771

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/72b992de-ddcf-4483-9690-31b2d0f53870

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/fa67a50a-724b-46ea-805f-e33ae81c958c

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/938046b9-865b-4882-bfae-0a02498ccb34

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/e450e4ba-0dad-4bf8-990a-96a5d06547c2

</details>

burczu committed:
enter email, hang tight and director check components added

burczu committed:
basic sub-steps components added

burczu committed:
signer info implementation

burczu committed:
name sub-step refactored to use common step component

burczu committed:
job title component refactored using common single step comp.

burczu committed:
refactor: showing footer component in date of birth step instead fixed help links

burczu committed:
date of birth using common step component

burczu committed:
confirmation refactored using common step component

burczu committed:
upload document ID's issue fixed

burczu committed:
prettier

dannymcclain commented:
I wasn't super involved with this, so we should probably wait for the other members of @Expensify/design to weigh in here, but _generally_ this looks pretty good. A few questions/comments:

- What size are the radio check circles? They look a little small to me? In Figma they're `28x28px`
- Seems like there's some weirdness going on with the attachment selector on iOS—any idea what's going on there?
- The attachment preview looks pretty borked when the file name is really long (icon gets pushed out of the container, the file name doesn't truncate, etc)
  - Is that something we need to handle here or is that being dealt with elsewhere?
- I think all the single input screens (legal name, job title) need to autofocus the input when you get to them (for the first time or any subsequent time)—that's how we typically handle inputs like that

shawnborton commented:
Feeling good! I have the same feedback as Danny above regarding the checkbox size. 

Then for this screen, the blue link looks like it's not left-aligned the same way the others are:
![CleanShot 2024-11-06 at 15 49 38@2x](https://github.com/user-attachments/assets/5a787852-5911-420d-bdad-2ba87536339f)

Then for this screen, it feels like the input could be a bit closer to the headline?

dubielzyk-expensify commented:
>Then for this screen, it feels like the input could be a bit closer to the headline?

This goes for the legal name too

hungvu193 commented:
@burczu would you mind addressing these advices from design team above? ty

burczu committed:
Merge branch 'main' into feat/50901-GR-step-5-director-information-screen

burczu commented:
@hungvu193 yes, on it!

burczu committed:
radio buttons styling fix

hungvu193 commented on src/pages/ReimbursementAccount/NonUSD/SignerInfo/DirectorCheck.tsx:
we had a similar component `BeneficialOwnerCheckUBO` can we replace it with this one?

burczu replied:
I think it would be better to extract the content of these components to some common one and use it in both places - what do you think?

hungvu193 replied:
I think that's ok

hungvu193 commented on src/pages/ReimbursementAccount/NonUSD/SignerInfo/EnterEmail.tsx:
```diff
@@ -0,0 +1,83 @@
+import INPUT_IDS from '@src/types/form/ReimbursementAccountForm';
+
+type EnterEmailProps = {
+    onSubmit: () => void;
+
+    isUserDirector: boolean;
```
Please add docs

burczu replied:
added

hungvu193 commented on src/pages/ReimbursementAccount/NonUSD/SignerInfo/EnterEmail.tsx:
```diff
@@ -0,0 +1,83 @@
+            }
+
+            if (shouldGatherBothEmails && values[SECOND_SIGNER_EMAIL] && !Str.isValidEmail(values[SECOND_SIGNER_EMAIL])) {
+                errors[SECOND_SIGNER_EMAIL] = translate('bankAccount.error.lastName');
```
SAme here

burczu replied:
fixed

hungvu193 commented:
Error message is to close to the Upload file component.

<img width="1700" alt="Screenshot 2024-11-11 at 10 40 08" src="https://github.com/user-attachments/assets/8fb6be48-9508-41be-80dd-a72845dd0ab2">

hungvu193 commented:
Infinite loop when pressing back between LegalName and Email page.

https://github.com/user-attachments/assets/e395993e-2b70-4cee-a108-5d6c3336d77b

hungvu193 commented:
On Native platform, the attachment icon is out of the border.

<img width="582" alt="Screenshot 2024-11-11 at 11 11 46" src="https://github.com/user-attachments/assets/0d49332f-3406-4f97-8837-7bb88d1876ed">

hungvu193 commented:
@burczu  Little bump for these comments above 👀

burczu committed:
Merge branch 'main' into feat/50901-GR-step-5-director-information-screen

burczu committed:
validation message for email address fixed

burczu commented on src/pages/ReimbursementAccount/NonUSD/SignerInfo/EnterEmail.tsx:
```diff
@@ -0,0 +1,83 @@
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.REIMBURSEMENT_ACCOUNT_FORM> => {
+            const errors = ValidationUtils.getFieldRequiredErrors(values, shouldGatherBothEmails ? [SIGNER_EMAIL, SECOND_SIGNER_EMAIL] : [SIGNER_EMAIL]);
+            if (values[SIGNER_EMAIL] && !Str.isValidEmail(values[SIGNER_EMAIL])) {
+                errors[SIGNER_EMAIL] = translate('bankAccount.error.firstName');
```
fixed

burczu committed:
missing docs comments added

burczu committed:
incorrect error message spacing of file upload fixed

burczu commented:
> Error message is too close to the Upload file component.

fixed

burczu committed:
back button infinite loop fixed

burczu committed:
Merge branch 'main' into feat/50901-GR-step-5-director-information-screen

burczu commented:
> Infinite loop when pressing back between LegalName and Email page.

fixed

burczu committed:
upload file styling fixed

burczu commented:
> On Native platform, the attachment icon is out of the border.

fixed

hungvu193 commented:
Nice. I'll verify it shortly

hungvu193 commented:
@burczu Can you also update the screenshots/videos so Design team can review the UI?

burczu commented:
> @burczu Can you also update the screenshots/videos so Design team can review the UI?

I will, but still I have few things here to address, so I'll do it once its all done. I think you can wait till then with verification - I'll post a separate comment that's it ready to re-review.

hungvu193 commented:
> > @burczu Can you also update the screenshots/videos so Design team can review the UI?
> 
> I will, but still I have few things here to address, so I'll do it once its all done. I think you can wait till then with verification - I'll post a separate comment that's it ready to re-review.

Cool. Let me know when it's ready for another review. Ty

burczu committed:
yes or no step component extracted

burczu committed:
bottom link styles fixed

burczu committed:
single field spacing fixed

burczu committed:
autofocus added

burczu commented:
@dannymcclain Answering your comments:

> I wasn't super involved with this, so we should probably wait for the other members of @Expensify/design to weigh in here, but _generally_ this looks pretty good. A few questions/comments:
> 
> * What size are the radio check circles? They look a little small to me? In Figma they're `28x28px`

Good point - fixed.

> * Seems like there's some weirdness going on with the attachment selector on iOS—any idea what's going on there?

I think this is because of the simulator, I believe this will work correctly on real device.

> * The attachment preview looks pretty borked when the file name is really long (icon gets pushed out of the container, the file name doesn't truncate, etc)
>   
>   * Is that something we need to handle here or is that being dealt with elsewhere?

This is related to what @hungvu193 [pointed out here](https://github.com/Expensify/App/pull/51648#issuecomment-2467195193) too - fixed.

> * I think all the single input screens (legal name, job title) need to autofocus the input when you get to them (for the first time or any subsequent time)—that's how we typically handle inputs like that

Added autofocus to sile input screens.

burczu commented:
@shawnborton Answering your comments:

> Feeling good! I have the same feedback as Danny above regarding the checkbox size.
> 
> Then for this screen, the blue link looks like it's not left-aligned the same way the others are: 

Good point. Fixed.

> Then for this screen, it feels like the input could be a bit closer to the headline? 

Yes. Fixed

burczu commented:
@hungvu193 I think I've addressed all the comments - you can now re-review the PR. Videos also are updated now.

hungvu193 commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->


https://github.com/user-attachments/assets/b235b38b-56d0-47f6-a823-da98ce17a2f9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->


https://github.com/user-attachments/assets/a828bd0f-5926-4c68-bdcf-6b325d7ecdc5

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/5805ff5c-e795-4a4f-bd27-6e52579c793d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/b9f803c4-e7f4-4721-aec5-2e6e270625ab

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/user-attachments/assets/dabf071e-70fb-4ede-9ae9-ca414dfde927

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->


https://github.com/user-attachments/assets/c5ba5b58-ba6e-4368-a465-fd3f36b9e920

</details>

hungvu193 commented:
@burczu  LGTM. I think we only have translations left

burczu committed:
Merge branch 'main' into feat/50901-GR-step-5-director-information-screen

burczu committed:
translations fixed

burczu commented:
@hungvu193 asked for confirmation on translations here: https://callstack-hq.slack.com/archives/C01GTK53T8Q/p1731572554005439

burczu committed:
spanish translations updated

burczu commented:
@hungvu193 spanish translations reviewed and updated

hungvu193 reviewed (approved):
LGTM

burczu committed:
Merge branch 'main' into feat/50901-GR-step-5-director-information-screen

madmax330 commented on src/pages/ReimbursementAccount/NonUSD/WhyLink.tsx:
```diff
@@ -0,0 +1,44 @@
+    containerStyles?: StyleProp<ViewStyle>;
+};
+
+function WhyLink({containerStyles}: WhyLinkProps) {
```
This component was already added in @pasyukevich's PR here: https://github.com/Expensify/App/pull/52054/files

hungvu193 replied:
Oh It was merged recently, I didn't notice it during this PR review process

burczu committed:
upload file style fix

burczu committed:
Merge branch 'main' into feat/50901-GR-step-5-director-information-screen

burczu committed:
Merge branch 'main' into feat/50901-GR-step-5-director-information-screen

chiragsalian commented:
@burczu @hungvu193 @madmax330, can someone test if this works as intended. QA mentioned there are unable to see director information screen - [Issue](https://github.com/Expensify/App/issues/52631)

I'm going to check this off the QA checklist to not block deploy because it looks like worst case the screen doesn't show up which is the same as production.

hungvu193 commented:
> @burczu @hungvu193 @madmax330, can someone test if this works as intended on staging. QA mentioned they are unable to see director information screen - [Issue](https://github.com/Expensify/App/issues/52631)
> 
> I'm going to check this off the QA checklist to not block deploy because it looks like worst case the screen doesn't show up which is the same as production.

Sorry for this inconvenience. We have a plan to support this feature in Debug mode. However, we missed that case when we implemented it on https://github.com/Expensify/App/pull/51667.
So we decided to implement it in our next PR (which is [in progress here](https://github.com/Expensify/App/pull/52384/files#diff-caebaa0f8f71cb86d9ccac19fbde0edeb695f33a078eae26a9132b5b453c85ceR237)).

So I can say https://github.com/Expensify/App/issues/52631 will be fixed once https://github.com/Expensify/App/pull/52384 is merged.
